validator/client/beacon-api/genesis.go

```diff
@@ -0,0 +1,70 @@
+//go:build use_beacon_api
+// +build use_beacon_api
+
+package beacon_api
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/rpc/apimiddleware"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+)
+
+func (c *beaconApiValidatorClient) waitForChainStart() (*ethpb.ChainStartResponse, error) {
+	genesis, err := c.getGenesis()
+	if err != nil {
+		return nil, err
+	}
+
+	genesisTime, err := strconv.ParseUint(genesis.Data.GenesisTime, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	chainStartResponse := &ethpb.ChainStartResponse{}
+	chainStartResponse.Started = true
+	chainStartResponse.GenesisTime = genesisTime
+
+	// Remove the leading 0x from the string before decoding it to bytes
+	genesisValidatorRoot, err := hex.DecodeString(genesis.Data.GenesisValidatorsRoot[2:])
+	if err != nil {
+		return nil, err
+	}
+	chainStartResponse.GenesisValidatorsRoot = genesisValidatorRoot
+
+	return chainStartResponse, nil
+}
+
+func (c *beaconApiValidatorClient) getGenesis() (*apimiddleware.GenesisResponseJson, error) {
+	resp, err := c.httpClient.Get(c.url + "/eth/v1/beacon/genesis")
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err = resp.Body.Close(); err != nil {
+			return
+		}
+	}()
```

Comment:
can we do this instead? This will allow us to introduce request timeout
```suggestion
func (c *beaconApiValidatorClient) getGenesis(ctx context) (*apimiddleware.GenesisResponseJson, error) {
        ctx, cancel := context.WithTimeout(ctx, c.timeout)
	defer cancel()
	
        req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.url + "/eth/v1/beacon/genesis", nil)
	if err != nil {
		return nil, err
	}
	
	resp, err := c.httpClient.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
```

Reply:
`c.httpClient` already defines a client that has a timeout given in the constructor.

Reply:
yeah its there but
https://stackoverflow.com/questions/64129364/go-http-client-timeout-vs-context-timeout
https://stackoverflow.com/questions/48900772/specify-timeout-when-tracing-http-request-in-go

Context timeout helps if context is propagated from parent functions and context gets cancelled or deadline is exceeded in parent goroutine.
Would love to hear thoughts on this from prysm team @rkapka @kasey